src/taipy/core/scenario/_scenario_manager.py

```diff
@@ -206,25 +208,27 @@ def _delete(cls, scenario_id: ScenarioId):  # type: ignore
         scenario = cls._get(scenario_id)
         if scenario.is_primary:
             if len(cls._get_all_by_cycle(scenario.cycle)) > 1:
-                raise DeletingPrimaryScenario
+                raise DeletingPrimaryScenario(
+                    f"Scenario {scenario.id}, which has config id {scenario.config_id}, is primary and there are other scenarios in the same cycle."
+                )
             _CycleManagerFactory._build_manager()._delete(scenario.cycle.id)
         super()._delete(scenario_id)
 
     @classmethod
     def _compare(cls, *scenarios: Scenario, data_node_config_id: str = None):
         if len(scenarios) < 2:
-            raise InsufficientScenarioToCompare
+            raise InsufficientScenarioToCompare("At least two scenarios are required to compare")
 
         if not all([scenarios[0].config_id == scenario.config_id for scenario in scenarios]):
-            raise DifferentScenarioConfigs
+            raise DifferentScenarioConfigs("Scenarios to compare must have the same configuration")
 
         if scenario_config := _ScenarioManager.__get_config(scenarios[0]):
             results = {}
             if data_node_config_id:
                 if data_node_config_id in scenario_config.comparators.keys():
                     dn_comparators = {data_node_config_id: scenario_config.comparators[data_node_config_id]}
                 else:
-                    raise NonExistingComparator
+                    raise NonExistingComparator(data_node_config_id)
```

Comment:
Please add context for end user.

Reply:
something like:
f"data node config {data_node_config_id} has no comparator."